src/components/tx/TxStepper/index.tsx

```diff
@@ -12,11 +12,15 @@ const TxStepper = ({ steps, initialData, initialStep, onClose }: TxStepperProps)
     onClose,
   })
 
+  const activeLabel = steps[activeStep].label
+
   return (
     <div className={css.container}>
       <ModalDialogTitle onClose={onClose}>
         <Grid container px={1} alignItems="center" gap={2}>
-          <Grid item>{steps[activeStep].label}</Grid>
+          <Grid item>
+            {typeof activeLabel === 'string' ? activeLabel : activeLabel(stepData[Math.max(0, activeStep)])}
```

Comment:
What's this for?

Reply:
This is in order to allow custom components on the modal header. We need it for showing the app icon with the app name (both dynamic)

Reply:
Done!